apps/src/lib/ui/ConfirmEnableMakerDialog.jsx

```diff
@@ -0,0 +1,68 @@
+import React, {Component, PropTypes} from 'react';
+import Portal from 'react-portal';
+import msg from '@cdo/locale';
+import color from '../../util/color';
+import Dialog, {Title, Body} from '../../templates/Dialog';
+
+const style = {
+  description: {
+    fontSize: 'smaller',
+  },
+  warning: {
+    color: color.red,
+    fontSize: 'smaller',
+    fontStyle: 'italic',
+    textAlign: 'center',
+    padding: 10,
+  },
+};
+
+export class ConfirmEnableMakerDialog extends Component {
+  static propTypes = {
+    isOpen: PropTypes.bool.isRequired,
+    handleConfirm: PropTypes.func.isRequired,
+    handleCancel: PropTypes.func.isRequired,
+  };
+
+  render() {
+    return (
+      <Dialog
+        isOpen={this.props.isOpen}
+        confirmText={msg.enable()}
+        onConfirm={this.props.handleConfirm}
+        cancelText={msg.dialogCancel()}
+        onCancel={this.props.handleCancel}
+        handleClose={this.props.handleCancel}
+      >
+        <Title>{msg.enableMakerDialogTitle()}</Title>
+        <Body>
+          <div style={style.description}>
+            {msg.enableMakerDialogDescription()}
+            {' '}
+            <a href="/maker/setup" target="_blank">
+              {msg.enableMakerDialogSetupPageLinkText()}
+            </a>
+          </div>
+          <div style={style.warning}>
+            {msg.enableMakerDialogWarning()}
+          </div>
+        </Body>
+      </Dialog>
+    );
+  }
+}
+
+// Our default export is actually a wrapper around our dialog that renders it
+// through a Portal component so it sits at the end of the DOM instead of
+// inside whatever component called for it - but this is lousy for testing,
+// so we mostly export and test the inner dialog component.
+export default class ConfirmEnableMakerDialogPortal extends Component {
```

Comment:
I think we should do this for all dialogs (to avoid stylesheet bleed) _but_ there are some nasty implications for testing and maybe for storybook, since enzyme can't get a handle on the content rendered through the portal (hence the divide here).  I'd also like to keep the footprint of this PR limited to this feature, and tackle a larger change affecting all dialogs as a separate PR.